Framework/Algorithms/inc/MantidAlgorithms/AbsorptionCorrection.h

```diff
@@ -113,11 +114,11 @@ class DLLExport AbsorptionCorrection : public API::Algorithm {
    */
   virtual void initialiseCachedDistances() = 0;
 
-  API::MatrixWorkspace_sptr m_inputWS;    ///< A pointer to the input workspace
-  const Geometry::Object *m_sampleObject; ///< Local cache of sample object.
-  Kernel::V3D m_beamDirection;            ///< The direction of the beam.
-  std::vector<double> m_L1s,              ///< Cached L1 distances
-      m_elementVolumes;                   ///< Cached element volumes
+  API::MatrixWorkspace_sptr m_inputWS; ///< A pointer to the input workspace
+  const Geometry::CSGObject *m_sampleObject; ///< Local cache of sample object.
```

Comment:
Will `AbsorptionCorrection` work with any object (`IObject`) or only with CSG?

Reply:
Not with IObject if it keeps a local cache of sample object.